plone/app/vocabularies/terms.py

```diff
@@ -25,7 +25,14 @@ def safe_simpleterm_from_value(value):
     - token need cleaned up: Vocabulary term tokens *must* be 7 bit values
     - anything for display has to be cleaned up, titles *must* be unicode
     """
-    return SimpleTerm(value, b2a_qp(safe_encode(value)), safe_unicode(value))
+    if isinstance(value, six.text_type):
```

Comment:
This *doesn't test for newlines*. The text string `"This is English text using ASCII characters\n"` is not a valid token, but will be used as a token nonetheless. So _at the very least_ this needs to verify that there are no line separators in the text.

I haven't done any timing tests, but either a regex test or `len(value.splitlines()) == 1` or `value == value.translate({10: None, 13: None})` is needed to test for any of the `\r`, `\r\n` and `\n` combinations.

In the end, it _probably_ is just simpler to use `urlsafe_b64encode(safe_encode(value))` and not bother with preserving ASCII-only values.

Reply:
using the simple path - using urlsafe_b64encode (which will never produce a newline) and doing away with the readability of tokens.